Core/include/Acts/Propagator/detail/JacobianEngine.hpp

```diff
@@ -0,0 +1,145 @@
+/// pathCorrectionFactor(gloB))*jacTransport(gloA->gloB) *jac(locA->gloA)
+///
+/// @return a 6x6 transport jacobian from bound to bound
+BoundMatrix boundToBoundTransportJacobian(
+    const GeometryContext& geoContext, const FreeVector& freeParameters,
+    const BoundToFreeMatrix& boundToFreeJacobian,
+    const FreeMatrix& freeTransportJacobian,
+    const FreeVector& freePathDerivatives, const Surface& surface);
+
+/// @brief This function calculates the full jacobian from a given
+/// bound/curvilinear parameterisation from a surface to new curvilinear
+/// parameterisation.
+///
+/// @note Modifications of the jacobian related to the
+/// projection onto a curvilinear surface is considered. Since a variation of
+/// the start parameters within a given uncertainty would lead to a variation of
+/// the end parameters, these need to be propagated onto the target surface.
+/// This an approximated approach to treat the (assumed) small change.
+///
+/// @param [in] direction Normalised direction vector
+/// @param [in] boundToFreeJacobian Jacobian from bound to free at start
+/// @param [in] freeTransportJacobian Transport jacobian free to free
+/// @param [in] freePathDerivatives Path length derivatives for free parameters
+///
+/// @note The parameter @p surface is only required if projected to bound
+/// parameters. In the case of curvilinear parameters the geometry and the
+/// position is known and the calculation can be simplified
+///
+/// @return a 6x6 transport jacobian from curilinear to bound
+BoundMatrix boundToCurvilinearTransportJacobian(
+    const Vector3& direction, const BoundToFreeMatrix& boundToFreeJacobian,
+    const FreeMatrix& freeTransportJacobian,
+    const FreeVector& freePathDerivatives);
+
+/// @brief This function calculates the full jacobian from a given
+/// bound/curvilinear parameterisation from a new free
+/// parameterisation.
+///
+/// @param [in] boundToFreeJacobian Jacobian from bound to free at start
+/// @param [in] freeTransportJacobian Transport jacobian free to free
+///
+/// @return a 8x6 transport jacobian from bound to free
+BoundToFreeMatrix boundToFreeTransportJacobian(
+    const BoundToFreeMatrix& boundToFreeJacobian,
+    const FreeMatrix& freeTransportJacobian);
+
+/// @brief This function calculates the full jacobian from a given
+/// free parameterisation to a new curvilinear bound
+/// parameterisation.
+///
+/// @note Modifications of the jacobian related to the
+/// projection onto the target surface is considered. Since a variation of
+/// the start parameters within a given uncertainty would lead to a variation of
+/// the end parameters, these need to be propagated onto the target surface.
+/// This an approximated approach to treat the (assumed) small change.
+///
+/// @param [in] geoContext The geometry Context
+/// @param [in] freeParameters Free, nominal parametrisation
+/// @param [in] angleToDirectionJacobian The relation jacobian from angle to dir
```

Comment:
`@param [in] directionToAngleJacobian` is missing

Reply:
Oh, I see - will fix it.